### ISSUE TITLE
FileNotFoundException at com.amplitude.core.utilities.EventsFileManager.writeToFile

### ISSUE THREAD

**Issue by shawn-lin-013** [amplitude/Amplitude-Kotlin#187]
Encountering numerous crashes in version 1.6.0.

```
Fatal Exception: java.io.FileNotFoundException: /data/user/0/package_name/app_amplitude-disk-queue/<redacted>-71975.tmp (Permission denied)
       at java.io.FileOutputStream.open0(FileOutputStream.java)
       at java.io.FileOutputStream.open(FileOutputStream.java:287)
       at java.io.FileOutputStream.<init>(FileOutputStream.java:223)
       at com.amplitude.core.utilities.EventsFileManager.writeToFile(EventsFileManager.kt:168)
       at com.amplitude.core.utilities.EventsFileManager.finish(EventsFileManager.kt:145)
       at com.amplitude.core.utilities.EventsFileManager.rollover(EventsFileManager.kt:104)
       at com.amplitude.android.utilities.AndroidStorage.rollover(AndroidStorage.kt:54)
       at com.amplitude.core.platform.EventPipeline$upload$1$1$1.invokeSuspend(EventPipeline.kt:102)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
```

**Comment by izaaz** [comment]
Thanks for reporting this @shawn-lin-013. Is this happening to a certain type of devices? Are you able to repro this issue locally?

**Comment by qingzhuozhen** [comment]
Hi @shawn-lin-013, we recently pushed some fixes around storage to resolve some crashes. Would you mind upgrading to the latest and see if that helps?

**Comment by shawn-lin-013** [comment]
> Hi @shawn-lin-013, we recently pushed some fixes around storage to resolve some crashes. Would you mind upgrading to the latest and see if that helps?

Hi @qingzhuozhen, we wonder which version is suggested to upgrade?

**Comment by qingzhuozhen** [comment]
Hi @shawn-lin-013 We recently released v1.16.4, maybe try to upgrade to that version and see if that helps. Feel free to let us know if the issue is still not solved and we can dive further,

**Comment by izaaz** [comment]
@shawn-lin-013 closing this issue but please feel free to reopen the issue if you still see problems with the latest version.
